app/components/edit/step_component.html.erb

```diff
@@ -1,12 +1,12 @@
-<%= tag.section class: classes, data:, id:, 'aria-labelledby': aria_labelledby do %>
+<%= tag.section class: classes, data:, id:, 'aria-label': aria_label do %>
   <div class="card-header card-step-header">
     <%= tag.div class: 'd-flex align-items-start' do %>
-      <div><%= render Edit::CharacterCircleComponent.new(character: step_number, classes: 'me-2', variant: character_circle_variant, id: character_circle_id) %></div>
+      <div><%= render Edit::CharacterCircleComponent.new(character: step_number, classes: 'me-2', variant: character_circle_variant) %></div>
       <div class="flex-grow-1">
         <div class="d-flex align-items-start">
-          <%= tag.h2 class: 'h4', id: title_id do %><%= title %><% end %>
-          <%= render Edit::InProgressBadgeComponent.new(classes: 'ms-3 mt-1', id: in_progress_badge_id) if show? %>
-          <%= render Edit::CompletedBadgeComponent.new(classes: 'ms-3 mt-1', id: completed_badge_id) unless show? %>
+        <h2 class="h4"> <%= title %> </h2>
+          <%= render Edit::InProgressBadgeComponent.new(classes: 'ms-3 mt-1') if show? %>
+          <%= render Edit::CompletedBadgeComponent.new(classes: 'ms-3 mt-1') unless show? %>
```

Comment:
Unrelated to the changes in this PR, I wonder if we might want to shift this logic into the Ruby side of the component. Maybe something like:


```suggestion
          <%= render badge_component.new(classes: badge_classes) %>
```

```ruby
  # in `app/components/edit/step_component.rb`
  def badge_component
    if show?
      Edit::InProgressBadgeComponent
    else
      Edit::CompletedBadgeComponent
    end
  end

  def badge_classes
    'ms-3 mt-1'
  end
```

❓ Not urgent or blocking.

Reply:
I like this. Going with a slight variation.